### ISSUE TITLE
Photos 2.0 Can't Delete photo from single view

### ISSUE THREAD

**Issue by yankees5963** [nextcloud/photos#1360]
I can delete a photo from the grid screen, but if i click into a photo which brings up the single photo view and click the delete button i get the following error:

AxiosError: Request failed with status code 404

Willing to provide any additional logs (didnt see anything relatable in server logs).

System:
Nextcloud 25 RC5
Docker Install
PHP8

**Comment by skjnldsv** [comment]
Hello!
I will need your browser console log to investigate this issue.
Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by yankees5963** [comment]
Console Log:

```
REDACTEDURL/:1 Refused to apply style from 'https://REDACTEDURL/apps/photos/css/icons.css?v=d233662f-3690' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled.
index.es.js:2337 Proxying an event bus of version 2.1.1 with 1.3.0
e @ index.es.js:2337
70903 @ index.es.js:3314
a @ bootstrap:19
71894 @ requesttoken.js:11
a @ bootstrap:19
22200 @ index.js:25
a @ bootstrap:19
12498 @ index.js:323
a @ bootstrap:19
(anonymous) @ startup:4
a.O @ chunk loaded:25
(anonymous) @ startup:5
(anonymous) @ startup:5
jquery-migrate.min.js:2 JQMIGRATE: Migrate is installed, version 3.4.0
index.es.js:2337 Proxying an event bus of version 2.1.1 with 1.3.0
e @ index.es.js:2337
70903 @ index.es.js:3314
i @ bootstrap:19
71894 @ requesttoken.js:11
i @ bootstrap:19
22200 @ index.js:25
i @ bootstrap:19
55058 @ LoggerBuilder.js:10
i @ bootstrap:19
17499 @ index.js:13
i @ bootstrap:19
15458 @ chunk loaded:1
i @ bootstrap:19
(anonymous) @ startup:4
i.O @ chunk loaded:25
(anonymous) @ startup:5
(anonymous) @ startup:5
globals.js:60 jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own.
$e @ globals.js:60
get @ globals.js:91
(anonymous) @ jquery.js:10853
(anonymous) @ jquery.js:28
19755 @ jquery.js:14
r @ bootstrap:19
38994 @ .*$:292
r @ bootstrap:19
(anonymous) @ startup:4
r.O @ chunk loaded:25
(anonymous) @ startup:5
(anonymous) @ startup:5
globals.js:60 $ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own.
$e @ globals.js:60
get @ globals.js:91
(anonymous) @ jquery.js:10856
(anonymous) @ jquery.js:28
19755 @ jquery.js:14
r @ bootstrap:19
38994 @ .*$:292
r @ bootstrap:19
(anonymous) @ startup:4
r.O @ chunk loaded:25
(anonymous) @ startup:5
(anonymous) @ startup:5
index.es.js:2337 Proxying an event bus of version 2.1.1 with 1.3.0
e @ index.es.js:2337
70903 @ index.es.js:3314
r @ bootstrap:19
71894 @ requesttoken.js:11
r @ bootstrap:19
22200 @ index.js:25
r @ bootstrap:19
4820 @ index.js:14
r @ bootstrap:19
38994 @ .*$:292
r @ bootstrap:19
(anonymous) @ startup:4
r.O @ chunk loaded:25
(anonymous) @ startup:5
(anonymous) @ startup:5
index.es.js:2337 Proxying an event bus of version 2.1.1 with 1.3.0
e @ index.es.js:2337
70903 @ index.es.js:3314
i @ bootstrap:19
71894 @ requesttoken.js:11
i @ bootstrap:19
22200 @ index.js:25
i @ bootstrap:19
(anonymous) @ Avatar.js:24
n @ Avatar.js:1
(anonymous) @ Avatar.js:151
n @ Avatar.js:1
(anonymous) @ Avatar.js:1
83507 @ Avatar.js:1
i @ bootstrap:19
1794 @ web.dom-collections.for-each.js:13
i @ bootstrap:19
85063 @ chunk loaded:1
i @ bootstrap:19
(anonymous) @ startup:4
i.O @ chunk loaded:25
(anonymous) @ startup:5
(anonymous) @ startup:5
major.js:2 Proxying an event bus of version 2.1.1 with 1.3.0
e @ major.js:2
17331 @ web.dom-collections.for-each.js:13
i @ bootstrap:19
(anonymous) @ Avatar.js:24
n @ Avatar.js:1
(anonymous) @ Avatar.js:151
n @ Avatar.js:1
(anonymous) @ Avatar.js:1
83507 @ Avatar.js:1
i @ bootstrap:19
1794 @ web.dom-collections.for-each.js:13
i @ bootstrap:19
85063 @ chunk loaded:1
i @ bootstrap:19
(anonymous) @ startup:4
i.O @ chunk loaded:25
(anonymous) @ startup:5
(anonymous) @ startup:5
index.js:19 Proxying an event bus of version 2.1.1 with 3.0.0
e @ index.js:19
79862 @ index.js:71
i @ bootstrap:19
62021 @ index.js:5
i @ bootstrap:19
3607 @ NcAvatar.js:2
a @ NcAvatar.js:2
3351 @ NcAvatar.js:2
a @ NcAvatar.js:2
(anonymous) @ NcAvatar.js:2
(anonymous) @ NcAvatar.js:2
75925 @ NcAvatar.js:2
i @ bootstrap:19
85063 @ chunk loaded:1
i @ bootstrap:19
(anonymous) @ startup:4
i.O @ chunk loaded:25
(anonymous) @ startup:5
(anonymous) @ startup:5
globals.js:60 Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20
$e @ globals.js:60
get @ globals.js:91
t.default @ no-conflict.js:8
86834 @ handlebars.runtime.js:60
l @ bootstrap:19
40202 @ runtime.js:3
l @ bootstrap:19
94702 @ item.handlebars:1
l @ bootstrap:19
22420 @ chunk loaded:1
l @ bootstrap:19
(anonymous) @ startup:4
l.O @ chunk loaded:25
(anonymous) @ startup:5
(anonymous) @ startup:5
ProxyBus.ts:16 Proxying an event bus of version 2.1.1 with 3.0.2
e @ ProxyBus.ts:16
74854 @ index.ts:27
a @ bootstrap:19
22200 @ index.js:5
a @ bootstrap:19
3607 @ NcAvatar.js:2
r @ NcAvatar.js:2
3351 @ NcAvatar.js:2
r @ NcAvatar.js:2
(anonymous) @ NcAvatar.js:2
(anonymous) @ NcAvatar.js:2
75925 @ NcAvatar.js:2
a @ bootstrap:19
(anonymous) @ FilesSidebarCallViewApp.vue?vue&type=template&id=15b2c996&scoped=true&:1
(anonymous) @ mainFilesSidebar.js:96
(anonymous) @ mainFilesSidebarLoader.js:89
ProxyBus.ts:16 Proxying an event bus of version 2.1.1 with 3.0.2
e @ ProxyBus.ts:16
78595 @ index.ts:27
a @ bootstrap:19
(anonymous) @ HandBackLeft.vue?55e6:19
(anonymous) @ mainFilesSidebar.js:96
(anonymous) @ mainFilesSidebarLoader.js:89
globals.js:60 jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own.
$e @ globals.js:60
get @ globals.js:91
(anonymous) @ ua-parser.js:858
42238 @ ua-parser.js:11
a @ bootstrap:19
29823 @ BlackVideoEnforcer.js:47
a @ bootstrap:19
66294 @ localmedia.js:11
a @ bootstrap:19
52160 @ webrtc.js:6
a @ bootstrap:19
3454 @ simplewebrtc.js:3
a @ bootstrap:19
97456 @ signaling.js:1410
a @ bootstrap:19
(anonymous) @ Grid.vue?7b9b:20
(anonymous) @ mainFilesSidebar.js:96
(anonymous) @ mainFilesSidebarLoader.js:89
ProxyBus.ts:11 Proxying an event bus of version 2.1.1 with 3.0.2
e @ ProxyBus.ts:11
(anonymous) @ ProxyBus.ts:59
(anonymous) @ main.js:81
(anonymous) @ main.js:81
index.es.js:2337 Proxying an event bus of version 2.1.1 with 1.3.0
e @ index.es.js:2337
8088 @ index.es.js:3314
r @ bootstrap:19
9517 @ requesttoken.js:11
r @ bootstrap:19
4273 @ index.js:25
r @ bootstrap:19
5058 @ LoggerBuilder.js:10
r @ bootstrap:19
7499 @ index.js:13
r @ bootstrap:19
(anonymous) @ logger.js:25
(anonymous) @ main.js:81
(anonymous) @ main.js:81
ConsoleLogger.js:60 [WARN] viewer: Some mimes were ignored because they are not enabled in the server previews config {level: 2, app: 'viewer', uid: 'will', ignoredMimes: Array(3)}
value @ ConsoleLogger.js:60
value @ ConsoleLogger.js:86
(anonymous) @ images.js:71
(anonymous) @ main.js:81
(anonymous) @ main.js:81
index.js:19 Proxying an event bus of version 2.1.1 with 3.0.2
e @ index.js:19
27594 @ index.js:71
a @ bootstrap:19
22200 @ index.js:5
a @ bootstrap:19
55058 @ LoggerBuilder.js:10
a @ bootstrap:19
17499 @ index.js:13
a @ bootstrap:19
52029 @ logger.js:3
a @ bootstrap:19
(anonymous) @ ViewerComponent.vue?5c5f:20
(anonymous) @ viewer.js:38
(anonymous) @ viewer.js:38
globals.js:60 oc_appswebroots is deprecated: use OC.appswebroots instead, this will be removed in Nextcloud 20
$e @ globals.js:60
get @ globals.js:91
74411 @ mime.js:63
a @ bootstrap:19
(anonymous) @ ViewerComponent.vue?5c5f:20
(anonymous) @ viewer.js:38
(anonymous) @ viewer.js:38
globals.js:60 oc_appswebroots is deprecated: use OC.appswebroots instead, this will be removed in Nextcloud 20
$e @ globals.js:60
get @ globals.js:91
74411 @ mime.js:63
a @ bootstrap:19
(anonymous) @ ViewerComponent.vue?5c5f:20
(anonymous) @ viewer.js:38
(anonymous) @ viewer.js:38
ProxyBus.ts:16 Proxying an event bus of version 2.1.1 with 3.0.2
e @ ProxyBus.ts:16
78595 @ index.ts:27
i @ bootstrap:19
45994 @ load script:2
i @ bootstrap:19
(anonymous) @ jsonp chunk loading:86
(anonymous) @ main.js:66
(anonymous) @ main.js:66
index.es.js:2337 Proxying an event bus of version 2.1.1 with 1.3.0
e @ index.es.js:2337
58088 @ index.es.js:3314
i @ bootstrap:19
69517 @ requesttoken.js:11
i @ bootstrap:19
32341 @ index.js:25
i @ bootstrap:19
55058 @ LoggerBuilder.js:10
i @ bootstrap:19
17499 @ index.js:13
i @ bootstrap:19
6115 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
9923 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
5606 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
5805 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
(anonymous) @ ncvuecomponents.js:2
(anonymous) @ ncvuecomponents.js:2
33476 @ ncvuecomponents.js:2
i @ bootstrap:19
(anonymous) @ Tag.vue?6116:19
(anonymous) @ main.js:66
(anonymous) @ main.js:66
ProxyBus.ts:16 Proxying an event bus of version 2.1.1 with 3.0.2
e @ ProxyBus.ts:16
74854 @ index.ts:27
i @ bootstrap:19
22200 @ index.js:5
i @ bootstrap:19
3607 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
3351 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
4208 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
9446 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
3432 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
9923 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
5606 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
5805 @ ncvuecomponents.js:2
a @ ncvuecomponents.js:2
(anonymous) @ ncvuecomponents.js:2
(anonymous) @ ncvuecomponents.js:2
33476 @ ncvuecomponents.js:2
i @ bootstrap:19
(anonymous) @ Tag.vue?6116:19
(anonymous) @ main.js:66
(anonymous) @ main.js:66
globals.js:60 jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own.
$e @ globals.js:60
get @ globals.js:91
(anonymous) @ script.js?v=e931bd8b-3690:492
globals.js:60 jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own.
$e @ globals.js:60
get @ globals.js:91
(anonymous) @ files.js?v=e931bd8b-3690:122
ProxyBus.ts:16 Proxying an event bus of version 2.1.1 with 3.0.2
e @ ProxyBus.ts:16
(anonymous) @ index.ts:27
(anonymous) @ main.js:37
(anonymous) @ main.js:37
ProxyBus.ts:16 Proxying an event bus of version 2.1.1 with 3.0.2
e @ ProxyBus.ts:16
4854 @ index.ts:27
a @ bootstrap:19
2200 @ index.js:5
a @ bootstrap:19
3607 @ NcUserBubble.js:2
a @ NcUserBubble.js:2
3351 @ NcUserBubble.js:2
a @ NcUserBubble.js:2
4208 @ NcUserBubble.js:2
a @ NcUserBubble.js:2
(anonymous) @ NcUserBubble.js:2
(anonymous) @ NcUserBubble.js:2
6122 @ NcUserBubble.js:2
a @ bootstrap:19
(anonymous) @ User.vue?vue&type=template&id=b064c4b8&scoped=true&:1
(anonymous) @ main.js:37
(anonymous) @ main.js:37
globals.js:60 $ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own.
$e @ globals.js:60
get @ globals.js:91
mounted @ NotificationsApp.vue:212
St @ vue.runtime.esm.js:2987
on @ vue.runtime.esm.js:4000
insert @ vue.runtime.esm.js:4391
T @ vue.runtime.esm.js:6897
Ba.nodeOps @ vue.runtime.esm.js:7108
e._update @ vue.runtime.esm.js:3734
r @ vue.runtime.esm.js:3844
e.get @ vue.runtime.esm.js:3415
e @ vue.runtime.esm.js:3405
e @ vue.runtime.esm.js:3861
Xn.$mount @ vue.runtime.esm.js:8727
e._init @ vue.runtime.esm.js:5656
Xn @ vue.runtime.esm.js:5717
(anonymous) @ main.js:33
(anonymous) @ main.js:37
(anonymous) @ main.js:37
globals.js:60 $ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own.
$e @ globals.js:60
get @ globals.js:91
mounted @ NotificationsApp.vue:218
St @ vue.runtime.esm.js:2987
on @ vue.runtime.esm.js:4000
insert @ vue.runtime.esm.js:4391
T @ vue.runtime.esm.js:6897
Ba.nodeOps @ vue.runtime.esm.js:7108
e._update @ vue.runtime.esm.js:3734
r @ vue.runtime.esm.js:3844
e.get @ vue.runtime.esm.js:3415
e @ vue.runtime.esm.js:3405
e @ vue.runtime.esm.js:3861
Xn.$mount @ vue.runtime.esm.js:8727
e._init @ vue.runtime.esm.js:5656
Xn @ vue.runtime.esm.js:5717
(anonymous) @ main.js:33
(anonymous) @ main.js:37
(anonymous) @ main.js:37
globals.js:60 $ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own.
$e @ globals.js:60
get @ globals.js:91
mounted @ NotificationsApp.vue:218
St @ vue.runtime.esm.js:2987
on @ vue.runtime.esm.js:4000
insert @ vue.runtime.esm.js:4391
T @ vue.runtime.esm.js:6897
Ba.nodeOps @ vue.runtime.esm.js:7108
e._update @ vue.runtime.esm.js:3734
r @ vue.runtime.esm.js:3844
e.get @ vue.runtime.esm.js:3415
e @ vue.runtime.esm.js:3405
e @ vue.runtime.esm.js:3861
Xn.$mount @ vue.runtime.esm.js:8727
e._init @ vue.runtime.esm.js:5656
Xn @ vue.runtime.esm.js:5717
(anonymous) @ main.js:33
(anonymous) @ main.js:37
(anonymous) @ main.js:37
NotificationsApp.vue:459 Notifications permissions not yet requested
session-heartbeat.js:103 session heartbeat polling started
index.js:19 Proxying an event bus of version 2.1.1 with 3.0.2
e @ index.js:19
9862 @ index.js:71
a @ bootstrap:19
2021 @ index.js:5
a @ bootstrap:19
2747 @ index.js:4
a @ bootstrap:19
768 @ NcRelatedResourcesPanel.js:2
r @ NcRelatedResourcesPanel.js:2
(anonymous) @ NcRelatedResourcesPanel.js:2
(anonymous) @ NcRelatedResourcesPanel.js:2
9234 @ NcRelatedResourcesPanel.js:2
a @ bootstrap:19
a.t @ create fake namespace object:10
Promise.then (async)
(anonymous) @ main.js:39
xhr.js:220          DELETE https://REDACTEDURL/remote.php/dav/files/will/files/will/CameraUploads/2022/10/20221014_115954.jpg 404
(anonymous) @ xhr.js:220
e.exports @ xhr.js:16
e.exports @ dispatchRequest.js:56
p.request @ Axios.js:109
r.forEach.p.<computed> @ Axios.js:131
(anonymous) @ bind.js:9
(anonymous) @ Viewer.vue:933
u @ runtime.js:63
(anonymous) @ runtime.js:294
(anonymous) @ runtime.js:119
Bg @ Pencil.vue?12b8:19
a @ Pencil.vue?12b8:19
(anonymous) @ Pencil.vue?12b8:19
(anonymous) @ Pencil.vue?12b8:19
onDelete @ Viewer.vue:928
Ot @ vue.runtime.esm.js:2987
n @ vue.runtime.esm.js:1785
Ot @ vue.runtime.esm.js:2987
e.$emit @ vue.runtime.esm.js:3685
onClick @ NcActionButton.js:2
Ot @ vue.runtime.esm.js:2987
n @ vue.runtime.esm.js:1785
Yr.i._wrapper @ vue.runtime.esm.js:7428
Viewer.vue:947 o {message: 'Request failed with status code 404', name: 'AxiosError', code: 'ERR_BAD_REQUEST', config: {…}, request: XMLHttpRequest, …}
(anonymous) @ Viewer.vue:947
u @ runtime.js:63
(anonymous) @ runtime.js:294
(anonymous) @ runtime.js:119
Bg @ Pencil.vue?12b8:19
s @ Pencil.vue?12b8:19
Promise.then (async)
Bg @ Pencil.vue?12b8:19
(anonymous) @ Pencil.vue?12b8:19
(anonymous) @ Pencil.vue?12b8:19
onDelete @ Viewer.vue:928
Ot @ vue.runtime.esm.js:2987
n @ vue.runtime.esm.js:1785
Ot @ vue.runtime.esm.js:2987
e.$emit @ vue.runtime.esm.js:3685
onClick @ NcActionButton.js:2
Ot @ vue.runtime.esm.js:2987
n @ vue.runtime.esm.js:1785
Yr.i._wrapper @ vue.runtime.esm.js:7428
```

**Comment by yankees5963** [comment]
in the console log it appears that files/{user} gets duplicated and is trying to delete /files/{user}/files/{user} which does not exist.

**Comment by skjnldsv** [comment]
Ah yes, right. 
Should be fixed alongside https://github.com/nextcloud/viewer/pull/1416

**Comment by simonspa** [comment]
@skjnldsv for some reason this still hits me with the NC25.0.2 photos app (listed as 2.0.1)

![Uploading Screenshot_20230107-134436.png…]()

**Comment by defulmere** [comment]
@simonspa same here, using NC 25.0.2 and Photos 2.0.1.

**Comment by vgraba** [comment]
I have the same problem and  it looks like it will only be fixed with NC 26

**Comment by umgfoin** [comment]
Resolved with [commit 2a41db5](https://github.com/nextcloud/viewer/commit/2a41db596d5ee715dc8ef563981f7ea3b6a09abf). 
See #1591 

Duplicate of #1591 (to be precise: vice versa ;-) )
++umgfoin.

**Comment by umgfoin** [comment]
> it looks like it will only be fixed with NC 26

Backported to upcoming 25.0.5
see #1591

**Comment by joshtrichards** [comment]
This issue can be closed out: https://github.com/nextcloud/photos/issues/1591#issuecomment-1516527019

Backported and merged to NC25 via  nextcloud/viewer#1561

**Comment by umgfoin** [comment]
Closed with [nextcloud/viewer/#1561](https://github.com/nextcloud/viewer/pull/1561)
